bin/mailmap_update.py

```diff
@@ -115,6 +120,41 @@ def author_name(line):
         for e in sorted(dups[k]):
             print('\t%s' % e)
 
+bad_names = []
+bad_emails = []
+for i in git_people:
+    name = author_name(i)
+    email = author_email(i)
+    if '@' in name:
+        bad_names.append(i)
+    elif '@' not in email:
```

Comment:
What about just putting all into one bad basket `bad_line`? If they make an error in either field they get flagged. The `if/elif` construct makes sure that if an error is made in either place that they get flagged. Then the error message could say "...appear to have the name and email reversed or an invalid email address. Entries...".

Reply:
Well bad emails is currently commented out, but if we decide to use it, they have different error messages. 

But I'm not really sure what to do about the email field. The motivation here was to catch people who got the email and name backwards when running `git config` (see the changes to `.mailmap` and `AUTHORS`), which currently checking for `@` in the name does. I left this here commented out in case we decide later to do something about it.

Reply:
I would go ahead and just enable the code below that is commented out: if they have the name/email reversed they will get flagged once as a bad name (because of the if/elif structure), get fixed and the problem should resolve on the next run. If they have a good name but a bad email, that will not get flagged (currently) unless you uncomment the code.

So by uncommenting the code below you will have no redundant listings and will be catching any possible missing email addresses.